simulationdataschema/model_method.py

```diff
@@ -34,19 +34,1646 @@
-from nomad.metainfo import Quantity, SubSection, SectionProxy, MEnum
+from nomad.datamodel.metainfo.annotations import ELNAnnotation
+from nomad.metainfo import (
+    Quantity,
+    SubSection,
+    MEnum,
+    SectionProxy,
+    Reference,
+    Section,
+    Context,
+    JSON,
+)
+
+from .model_system import ModelSystem
+from .atoms_state import OrbitalsState, CoreHole
+
+
+class NumericalSettings(ArchiveSection):
+    """
+    A base section used to define the numerical settings used in a simulation. These are meshes,
+    self-consistency parameters, and algebraic basis sets.
+    """
+
+    name = Quantity(
+        type=str,
+        description="""
+        Name of the numerical settings section. This is typically used to easy identification of the
+        `NumericalSettings` section. Possible values: "KMesh", "FrequencyMesh", "TimeMesh",
+        "SelfConsistency", "BasisSet".
+        """,
+        a_eln=ELNAnnotation(component="StringEditQuantity"),
+    )
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+
+class Mesh(NumericalSettings):
+    """
+    A base section used to specify the settings of a sampling mesh. It supports uniformly-spaced
+    meshes and symmetry-reduced representations.
+    """
+
+    sampling_method = Quantity(
+        type=MEnum(
+            "Gamma-centered",
+            "Monkhorst-Pack",
+            "Gamma-offcenter",
+            "Line-path",
+            "Equidistant",
+            "Logarithmic",
+            "Tan",
+            "Gauss-Legendre",
+            "Gauss-Laguerre" "Clenshaw-Curtis",
+            "Newton-Cotes",
+            "Gauss-Hermite",
+        ),
+        description="""
+        Method used to generate the mesh:
+
+        | Name      | Description                      |
+        | --------- | -------------------------------- |
+        | `'Gamma-centered'` | Regular mesh is centered around Gamma. No offset. |
+        | `'Monkhorst-Pack'` | Regular mesh with an offset of half the reciprocal lattice vector. |
+        | `'Gamma-offcenter'` | Regular mesh with an offset that is neither `'Gamma-centered'`, nor `'Monkhorst-Pack'`. |
+        | `'Line-path'` | Line path along high-symmetry points. Typically employed for simualting band structures. |
+        | `'Equidistant'`  | Equidistant 1D grid (also known as 'Newton-Cotes') |
+        | `'Logarithmic'`  | log distance 1D grid |
+        | `'Tan'`  | Non-uniform tan mesh for 1D grids. More dense at low abs values of the points, while less dense for higher values |
+        | `'Gauss-Legendre'` | Quadrature rule for integration using Legendre polynomials |
+        | `'Gauss-Laguerre'` | Quadrature rule for integration using Laguerre polynomials |
+        | `'Clenshaw-Curtis'`  | Quadrature rule for integration using Chebyshev polynomials using discrete cosine transformations |
+        | `'Gauss-Hermite'`  | Quadrature rule for integration using Hermite polynomials |
+        """,
```

Comment:
@JosePizarro3 ?

Reply:
Ok, I defined (just copy-pasting it here for simplicity):
```python
    spacing = Quantity(
        type=MEnum('Equidistant', 'Logarithmic', 'Tan'),
        default='Equidistant',
        description="""
        Identifier for the spacing of the Mesh. Defaults to 'Equidistant' if not defined. It can take the values:

        | Name      | Description                      |
        | --------- | -------------------------------- |
        | `'Equidistant'`  | Equidistant grid (also known as 'Newton-Cotes') |
        | `'Logarithmic'`  | log distance grid |
        | `'Tan'`  | Non-uniform tan mesh for grids. More dense at low abs values of the points, while less dense for higher values |
        """,
    )

    center = Quantity(
        type=MEnum('Gamma-centered', 'Monkhorst-Pack', 'Gamma-offcenter'),
        description="""
        Identifier for the center of the Mesh:

        | Name      | Description                      |
        | --------- | -------------------------------- |
        | `'Gamma-centered'` | Regular mesh is centered around Gamma. No offset. |
        | `'Monkhorst-Pack'` | Regular mesh with an offset of half the reciprocal lattice vector. |
        | `'Gamma-offcenter'` | Regular mesh with an offset that is neither `'Gamma-centered'`, nor `'Monkhorst-Pack'`. |
        """,
    )

    quadrature = Quantity(
        type=MEnum(
            'Gauss-Legendre',
            'Gauss-Laguerre',
            'Clenshaw-Curtis',
            'Newton-Cotes',
            'Gauss-Hermite',
        ),
        description="""
        Quadrature rule used for integration of the Mesh. This quantity is relevant for 1D meshes:

        | Name      | Description                      |
        | --------- | -------------------------------- |
        | `'Gauss-Legendre'` | Quadrature rule for integration using Legendre polynomials |
        | `'Gauss-Laguerre'` | Quadrature rule for integration using Laguerre polynomials |
        | `'Clenshaw-Curtis'`  | Quadrature rule for integration using Chebyshev polynomials using discrete cosine transformations |
        | `'Gauss-Hermite'`  | Quadrature rule for integration using Hermite polynomials |
        """,
    )
```

Reply:
Looking nice!